### PR TITLE
Grant the assistant entrypoints to every role and MCP administration to the admin roles

### PR DESCRIPTION
**Pull request by louisburroughs**
## Capability & Traceability
- Capability: _n/a — requested directly; no capability/story id supplied_
- Parent STORY (durion): _n/a_
- Child issue: _n/a_
- Domain: `domain:security`

## Contract References
- Contract guide entry (durion): `domains/security/.business-rules/BACKEND_CONTRACT_GUIDE.md` → role/permission resolution

No contract-relevant files changed — the diff is one Flyway migration, two tests and a README. Marker retained so the Contract Sync check has what it looks for.

## Contract Chain (when a Controller changed)
- [ ] OpenAPI annotations — no controller changed
- [ ] `openapi.yaml` regenerated — n/a
- [ ] Angular SDK regenerated — n/a

## Scope

**Every role** now receives the four conversational entrypoints:

| Permission | Bit | Meaning |
| --- | --- | --- |
| `mcp:chat:execute` | 226 | Synchronous chat via the Spring AI assistant runtime |
| `mcp:chat:stream` | 225 | Streaming SSE chat |
| `nlti:request:submit` | 221 | Submit a natural-language task-interface request |
| `nlti:request:read` | 222 | Read submitted NLTI request status |

**`ADMIN` and `SYSTEM_ADMINISTRATOR`** additionally receive MCP administration. The two wildcards were expanded from the manifest, not guessed:

- `mcp:system_prompt:*` → `view` (99), `create` (100), `update` (101), `delete` (102)
- `mcp:llm_api:*` → `view` (95), `create` (96), `update` (97), `delete` (98)
- `mcp:tool:manage` (347), `mcp:document:ingest` (224)

**`ADMIN`** also gets `mcp:tool:view` (348) and `mcp:tool:manage` — neither was held by any role before.

All sixteen names were verified present in a `permissions.yaml` manifest **and** in `PermissionCode` before use, so every one carries a bit index and can actually travel in a `perm_bits` claim. Grants go 603 → 674; roles 10 → 18.

## Three things worth your explicit attention

**1. `SYSTEM_ADMIN` isn't a role.** No such name exists. The seeded roles are `SYSTEM_ADMINISTRATOR` (the canonical persona this baseline already treats as the security/admin role) and `SECURITY_ADMIN` (a V3 "candidate role" that has never held anything). I applied the MCP administration set to **`SYSTEM_ADMINISTRATOR`**. Say the word if you meant `SECURITY_ADMIN`.

**2. External customers can now reach the assistant.** "Every role" includes `CUSTOMER` and `SELF_SERVICE_CUSTOMER`, which previously held **nothing at all**. They now hold `nlti:request:submit` — submitting natural-language task-interface requests — and both chat entrypoints. This is a real change in external attack surface, so I want it visible rather than buried in a diff.

My read is that it is defensible: these are reach-to-the-assistant permissions, and the assistant still enforces domain permissions per request, so a role that cannot read an invoice cannot read one by asking. But it rests on that enforcement actually holding for NLTI, which is outside what this PR verifies. If you want them excluded it is a two-line change to the seed plus the matching test.

**3. `SYSTEM_ADMINISTRATOR` gets `mcp:tool:manage` without `mcp:tool:view`.** That is exactly what was specified, so I implemented it literally rather than quietly widening it — but manage-without-view looks like an oversight. One line to add if you want it.

## Roles

Eighteen now, up from ten. The eight additions are the previously-empty roles: `CUSTOMER`, `SELF_SERVICE_CUSTOMER`, `SHOP_MANAGER`, `SECURITY_ADMIN`, `READ_ONLY_SCHEDULER`, and the three `INVENTORY_*`.

The `INVENTORY_*` three exist only via `RoleInitializer`, an `@PostConstruct` bean that runs **after** Flyway — the same ordering trap that broke main in #1372. The seed therefore creates them itself, exactly as it already creates `MANAGER` and `GENERAL_MANAGER`. Granting to them without that would abort startup.

The six unreachable legacy personas (`ACCOUNTANT`, `AP_CLERK`, `CONTROLLER`, `CSR`, `FLEET_MANAGER`, `GL_ANALYST`) remain excluded: nothing creates them and both assignment tables are FK'd to `roles(id)`, so "every role" cannot sensibly include roles no user can hold. Creating them is the open question in #1373.

## Tests
- [x] Unit tests added/updated
- [x] Integration tests updated

Three existing assertions became wrong **by design**. I updated them rather than deleting them, and each change narrows what is still guaranteed:

- **Legacy parity moves from equality to containment.** The baseline now deliberately exceeds the retired switch's expansion, so exact match is no longer the right assertion. The fixture stays as the no-regression floor — a capability the switch granted can still never silently disappear.
- **The `SYSTEM_ADMINISTRATOR` scope assertion** now admits `mcp:*` and `nlti:*` while still forbidding domain authority, and still requires a strict subset of `ADMIN`.
- **`DISPATCHER`'s exact-set assertion** gained the three new entrypoints.

New tests pin: the baseline on every role, `ADMIN`'s two tool permissions, and that MCP administration reaches **no** role other than the two admin ones.

`RolePermissionSeedIT` needed two fixes that matter more than they look — it is the test that only runs post-merge, so getting these wrong would have reddened main a second time:

- it asserted `CUSTOMER` held nothing, which is now false;
- it proved fail-closed *through* `CUSTOMER`, which no longer proves anything. Fail-closed is now demonstrated against a role created with no grants, so the property is genuinely tested rather than incidentally true.

**How to run:**

```bash
./mvnw -pl pos-security-service -am -DskipTests=false verify
```

Local result: **BUILD SUCCESS** — 479 unit + 82 IT, 0 failures, all quality gates green.

I also verified the generated SQL directly, independent of the tests: 18 roles, none missing the universal four, `ADMIN` holds both tool permissions, `SYSTEM_ADMINISTRATOR` holds the full MCP administration set, no leakage of MCP administration to any other role, and `SYSTEM_ADMINISTRATOR ⊂ ADMIN` still holds.

> As with every PR here, `RolePermissionSeedIT` will **not** run on this PR's CI — the `pull_request` path passes `-DskipITs` (`ci.yml:255-277`). It runs on the post-merge main build. I'll watch that run and fix forward if it fails.

## Risk & Rollback

**Risk level: Medium** — purely additive to authority, but it widens what customer-facing roles can reach (see point 2).

**Rollback:** revert the commit. Note the seed only inserts, so rows already applied persist after a revert; removing them means a `DELETE` through the admin API or a versioned migration.

## Checklist
- [ ] Branch name matches `cap/<cap-id>` — no cap id
- [ ] PR title starts with `[CAP:<cap-id>]` — happy to retitle
- [ ] Links to parent + child issues — none supplied
- [x] Contract guide updated — no contract semantics changed
- [ ] Required CI checks passing — pending

---
_Generated by [Claude Code](https://claude.ai/code/session_01HRTZc5gwTT869MoN8ga4j9)_

### CODE REVIEW

File: pos-security-service/src/test/java/com/positivity/securityservice/internal/config/RolePermissionBaselineTest.java [hunk 1]
```diff
@@ -218,9 +238,33 @@ void permissionDefinitionsMatchTheCatalogBitIndexes() throws IOException {
...
+
+        Set<String> holders = seededGrants.entrySet().stream()
+                .filter(entry -> entry.getValue().stream().anyMatch(MCP_ADMINISTRATION::contains))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toCollection(TreeSet::new));
+
+        assertThat(holders)
+                .as("MCP administration must not leak to operational or customer-facing roles")
+                .containsExactly("ADMIN", "SYSTEM_ADMINISTRATOR");
+    }
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-security-service/src/test/java/com/positivity/securityservice/internal/config/RolePermissionBaselineTest.java]: Good catch, and a real hole rather than a style point — fixed in 61435a2.

`MCP_ADMINISTRATION` is the set both admin roles must *hold*, and `mcp:tool:view` isn't in it because ADMIN holds it alone. But the restriction applies to it just the same, so scanning only that set meant `mcp:tool:view` could have been granted to any operational or customer-facing role with nothing failing, while the README asserted it was restricted. The leak scan now covers the full restricted surface (`MCP_ADMINISTRATION_SURFACE`), kept as a separate constant precisely because "what both admin roles must hold" and "what no other role may hold" are different sets here.

Verified rather than assumed: granting `mcp:tool:view` to TECHNICIAN now fails `mcpAdministrationIsReservedToAdminRoles`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pos-security-service/src/main/resources/db/migration/R__seed_role_permissions.sql [hunk 2]
```diff
@@ -8,26 +8,26 @@
...
--- * Roles intentionally left with zero grants: CUSTOMER, SELF_SERVICE_CUSTOMER,
---   SHOP_MANAGER, SECURITY_ADMIN, READ_ONLY_SCHEDULER, INVENTORY_LEAD,
---   INVENTORY_MANAGER, INVENTORY_CONTROLLER. They had none under the hardcoded
---   expansion either; adding grants is a product decision, not a migration.
+-- * Assistant baseline: EVERY role that exists receives mcp:chat:execute,
+--   mcp:chat:stream, nlti:request:submit and nlti:request:read. These are the
+--   conversational entrypoints; the assistant still enforces domain permissions
+--   per request, so holding them grants reach to the assistant, not to data.
+--   NOTE this includes the customer-facing CUSTOMER and SELF_SERVICE_CUSTOMER
+--   roles, which previously held nothing at all.
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-security-service/src/main/resources/db/migration/R__seed_role_permissions.sql]: Agreed, fixed in 61435a2 — the header now says "every role listed in section 3" and spells out the consequence.

The distinction is operationally significant, not just wording: this is a list of explicit grants, not a rule the database enforces. A role created later through `POST /v1/roles` or the role-permission admin API starts with **no** grants at all, assistant entrypoints included, until something grants them. The comment now says exactly that, and says to add the role to section 3 to make it part of the baseline.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: pos-security-service/README.md [hunk 3]
```diff
@@ -74,10 +74,33 @@ therefore creates those two roles itself rather than relying on bean lifecycle.
...
+| `nlti:request:submit` | Submit a natural-language task-interface request |
+| `nlti:request:read` | Read submitted NLTI request status |
+
+These grant reach to the assistant, not to data: the assistant enforces domain permissions per
+request, so a role that cannot read an invoice still cannot read one by asking for it. They are
+applied to **all** roles, including the customer-facing `CUSTOMER` and `SELF_SERVICE_CUSTOMER`,
+which previously held nothing at all — so external self-service users can now reach the assistant
+and submit NLTI requests. If that is not wanted, remove those two roles from the universal list in
+the seed; `RolePermissionBaselineTest.everyRoleReceivesTheAssistantBaseline` pins the current
+policy and will need updating alongside.
```
[the remark this answers is not in the document]
Comment (louisburroughs) [reply pos-security-service/README.md]: Agreed, fixed in 61435a2 — same correction as the SQL header.

The section now says "every role in the baseline seed" and adds the operational consequence explicitly: this is a list of explicit grants, not a rule the database enforces, so a role created later through `POST /v1/roles` or the role-permission admin API starts with no grants at all — assistant entrypoints included — until something grants them, and must be added to the seed to become part of the baseline.

Worth stating plainly because it interacts with the fail-closed behaviour documented just above: a newly created role carrying zero authority is the intended outcome, not a gap.

---
_Generated by [Claude Code](https://claude.ai/code)_
